submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestReleaser.kt

```diff
@@ -1,37 +1,57 @@
 package ac.uk.ebi.biostd.submission.submitter.request
 
 import ac.uk.ebi.biostd.persistence.common.model.RequestStatus.PROCESSED
+import ac.uk.ebi.biostd.persistence.common.model.SubmissionRequest
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceService
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestFilesPersistenceService
 import ac.uk.ebi.biostd.persistence.common.service.SubmissionRequestPersistenceService
 import ac.uk.ebi.biostd.persistence.filesystem.api.FileStorageService
 import ebi.ac.uk.extended.model.ExtFile
 import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.FireFile
 import mu.KotlinLogging
 import uk.ac.ebi.events.service.EventsPublisherService
 import uk.ac.ebi.extended.serialization.service.ExtSerializationService
 import uk.ac.ebi.extended.serialization.service.fileSequence
 
 private val logger = KotlinLogging.logger {}
 
+@Suppress("LongParameterList")
 class SubmissionRequestReleaser(
     private val fileStorageService: FileStorageService,
     private val serializationService: ExtSerializationService,
     private val eventsPublisherService: EventsPublisherService,
     private val queryService: SubmissionPersistenceQueryService,
     private val persistenceService: SubmissionPersistenceService,
     private val requestService: SubmissionRequestPersistenceService,
+    private val filesRequestService: SubmissionRequestFilesPersistenceService,
 ) {
+
     /**
      * Check the release status of the submission and release it if released flag is true.
      */
     fun checkReleased(accNo: String, version: Int): ExtSubmission {
         val request = requestService.getFilesCopiedRequest(accNo, version)
-        val sub = queryService.getExtByAccNoAndVersion(accNo, version, includeFileListFiles = true)
-        if (sub.released) releaseSubmission(sub)
-        requestService.updateRequestStatus(sub.accNo, sub.version, PROCESSED)
+        if (request.submission.released) releaseRequest(request)
+        requestService.updateRequestStatus(request.submission.accNo, request.submission.version, PROCESSED)
         eventsPublisherService.submissionSubmitted(accNo, request.notifyTo)
-        return sub
+        return request.submission
```

Comment:
I don't think it makes sense to return the submission anymore since we're not actually modifying it but just iterating through its files

Reply:
it help to know to the caller the last status of the extended error. For example check this code:
```
  fun handleRequest(accNo: String, version: Int): ExtSubmission {
        return when (requestService.getRequestStatus(accNo, version)) {
            REQUESTED -> completeRequest(accNo, version)
            INDEXED -> loadRequestFiles(accNo, version)
            LOADED -> cleanRequestFiles(accNo, version)
            CLEANED -> processRequestFiles(accNo, version)
            FILES_COPIED -> checkReleased(accNo, version)
            else -> throw IllegalStateException("Request accNo=$accNo, version='$version' has been already processed")
        }
    }
```

If we stop retrieving the ext submission alternative will be like quering it aside which wiill just make code less clear. Same with retrigger request endpoint as we want to know the final result if only release is pending we will need to query it asside.